homeassistant/components/climate/hive.py

```diff
@@ -0,0 +1,224 @@
+"""
+Support for the Hive devices.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/hive/
+"""
+import logging
+from datetime import datetime
+
+from homeassistant.components.climate import (ClimateDevice,
+                                              STATE_AUTO, STATE_HEAT,
+                                              STATE_OFF, STATE_ON)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+from homeassistant.components.hive import DATA_HIVE
+
+DEPENDENCIES = ['hive']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, hivedevice, discovery_info=None):

```

Comment:
Don't change the signature of the function. The fourth argument should be `discovery_info`. You can pass `hivedevice` as discovery info when you call `discovery.load_platform` from the hive component.

Reply:
Sorry, but I don't understand

Reply:
Define the function like this:
```py
def setup_platform(hass, config, add_devices, discovery_info=None):
```
Then you access `discovery_info` that reference the hivedevice dictionary you passed in from the component.

Reply:
please can you detail how to get hivedevice out of the discovery_info? looking through many of the platforms and I cant find one that does this

Reply:
They will be the same, so just `discovery_info`, where you used to use `hivedevice`.

Reply:
thanks. done